### PR TITLE
Ikasan 1706 module security 2

### PR DESCRIPTION
**Pull request by mick-stewart73**

### CODE REVIEW

File: ikasaneip/security/db/src/main/java/org/ikasan/security/model/RoleModule.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    /**
+     * The data time stamp when an instance was last updated
+     */
+    private Date updatedDateTime;
+
+    public RoleModule()
+    {
+        long now = System.currentTimeMillis();
+        this.createdDateTime = new Date(now);
```
Comment (mitcje) [inline ikasaneip/security/db/src/main/java/org/ikasan/security/model/RoleModule.java]: Why create Date objects? Why not just leave as longs?

File: ikasaneip/visualisation/dashboard/pom.xml [hunk 2]
```diff
@@ -233,6 +233,11 @@
             <artifactId>h2</artifactId>
         </dependency>
 
+        <dependency>
```
Comment (mitcje) [inline ikasaneip/visualisation/dashboard/pom.xml]: Why is the mysql connector in here?

File: ikasaneip/security/db/src/main/java/org/ikasan/security/model/RoleModule.java (hunk 1 above)
Comment (mick-stewart73) [reply ikasaneip/security/db/src/main/java/org/ikasan/security/model/RoleModule.java]: This has been the convention used on all security entities up to this point. If I change this one i will need to change all. Would you like me to go ahead?

File: ikasaneip/visualisation/dashboard/pom.xml (hunk 2 above)
Comment (mick-stewart73) [reply ikasaneip/visualisation/dashboard/pom.xml]: sorted

File: ikasaneip/security/db/src/main/java/org/ikasan/security/dao/HibernateSecurityDao.java [hunk 3]
```diff
@@ -329,6 +329,12 @@ public void deletePolicy(Policy policy)
         this.getHibernateTemplate().delete(policy);
     }
 
+    @Override
```
Comment (andrzej-majewski) [inline ikasaneip/security/db/src/main/java/org/ikasan/security/dao/HibernateSecurityDao.java]: How come there is only Delete  RoleModule, no add ?

File: ikasaneip/security/db/src/main/java/org/ikasan/security/model/RoleModule.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
+package org.ikasan.security.model;
+
+import java.util.Date;
+
+public class RoleModule
```
Comment (andrzej-majewski) [inline ikasaneip/security/db/src/main/java/org/ikasan/security/model/RoleModule.java]: Given you storing RoleModule objects in a set you need a equals and hashCode implementation.

Comment (mick-stewart73) [reply ikasaneip/security/db/src/main/java/org/ikasan/security/model/RoleModule.java]: done

File: ikasaneip/security/db/src/main/java/org/ikasan/security/dao/HibernateSecurityDao.java (hunk 3 above)
Comment (mick-stewart73) [reply ikasaneip/security/db/src/main/java/org/ikasan/security/dao/HibernateSecurityDao.java]: as discussed @andrzej-majewski we do not explicitly need to save RoleModule entities.

**Pull request merged**